homeassistant/components/anthemav/media_player.py

```diff
@@ -111,106 +118,77 @@ class AnthemAVR(MediaPlayerEntity):
-        self._attr_name = name
-        self._attr_unique_id = mac_address
+        self._zone_number = zone_number
+        self._zone = avr.zones[zone_number]
+        if zone_number > 1:
+            self._attr_name = f"zone {zone_number}"
+            self._attr_unique_id = f"{mac_address}_{zone_number}"
+        else:
+            self._attr_unique_id = mac_address
+
         self._attr_device_info = DeviceInfo(
             identifiers={(DOMAIN, mac_address)},
             name=name,
             manufacturer=MANUFACTURER,
             model=model,
         )
-
-    def _lookup(self, propname: str, dval: Any | None = None) -> Any | None:
-        return getattr(self.avr.protocol, propname, dval)
+        self._attr_device_class = MediaPlayerDeviceClass.RECEIVER
+        self._attr_icon = "mdi:audio-video"
+        self.set_states()
 
     async def async_added_to_hass(self) -> None:
         """When entity is added to hass."""
         self.async_on_remove(
             async_dispatcher_connect(
                 self.hass,
                 f"{ANTHEMAV_UDATE_SIGNAL}_{self._entry_id}",
-                self.async_write_ha_state,
+                self.update_states,
             )
         )
 
-    @property
-    def state(self) -> str | None:
-        """Return state of power on/off."""
-        pwrstate = self._lookup("power")
-
-        if pwrstate is True:
-            return STATE_ON
-        if pwrstate is False:
-            return STATE_OFF
-        return None
-
-    @property
-    def is_volume_muted(self) -> bool | None:
-        """Return boolean reflecting mute state on device."""
-        return self._lookup("mute", False)
-
-    @property
-    def volume_level(self) -> float | None:
-        """Return volume level from 0 to 1."""
-        return self._lookup("volume_as_percentage", 0.0)
-
-    @property
-    def media_title(self) -> str | None:
-        """Return current input name (closest we have to media title)."""
-        return self._lookup("input_name", "No Source")
-
-    @property
-    def app_name(self) -> str | None:
-        """Return details about current video and audio stream."""
-        return (
-            f"{self._lookup('video_input_resolution_text', '')} "
-            f"{self._lookup('audio_input_name', '')}"
-        )
-
-    @property
-    def source(self) -> str | None:
-        """Return currently selected input."""
-        return self._lookup("input_name", "Unknown")
-
-    @property
-    def source_list(self) -> list[str] | None:
-        """Return all active, configured inputs."""
-        return self._lookup("input_list", ["Unknown"])
+    def update_states(self) -> None:
+        """Update states for the current zone."""
+        self.set_states()
+        self.async_write_ha_state()
+
+    def set_states(self) -> None:
+        """Set all the states from the device to the entity."""
+        self._attr_state = STATE_ON if self._zone.power is True else STATE_OFF
```

Comment:
Can `self._zone.power` be something different than a boolean, eg `None`?

Reply:
not at this stage. If the zone is detected and created, the power is initialized with correct value. If we loose connection to the receiver, we retry connection but we don't set the state to unavailable. I might add this feature in the future, maybe to make the receiver unavailable after multiple unsuccessful retry connection.

Reply:
Ok.